content/articles/working-with-java-synchronization/index.md

```diff
@@ -0,0 +1,299 @@
+                try{
+                    Thread.sleep(400);
+                }catch(Exception e){System.out.println(e);}
+            }
+        }
+    }//end of the method
+}
+
+class MyThread1 extends Thread{
+    Table t;
+    MyThread1(Table t){
+        this.t=t;
+    }
+    public void run(){
+        t.printTable(5);
+    }
+
+}
+class MyThread2 extends Thread{
+    Table t;
+    MyThread2(Table t){
+        this.t=t;
+    }
+    public void run(){
+        t.printTable(100);
+    }
+}
+
+public class TestSynchronizedBlock1{
+    public static void main(String[] args){
+        Table obj = new Table();//only one object
+        MyThread1 t1=new MyThread1(obj);
+        MyThread2 t2=new MyThread2(obj);
+        t1.start();
+        t2.start();
+    }
+}
+```
+The above code will output the following
+
+```
+5
+10
+15
+20
+25
+100
+200
+300
+400
+500
+```
+
+The following is all that it entails:
+
+- The synchronized keyword is used to identify blocks that are part of a synchronized thread in Java. In Java, a synchronized block is one that is tied to a specific object. There can only be one thread operating in all synchronized blocks synchronized on the same object. When the synchronized block is exited, all subsequent threads trying to enter it are stalled until that thread quits.
+- Synchronized blocks utilize the synchronized keyword as a lock. When a method is marked as synchronized, the thread owns the monitor or lock object. In this case, you are blocked until the other thread releases the monitor.
+- Using a synchronized block enables you to fine-tune lock control by mutually excluding important section code.
+- The lock gets unlocked when the thread leaves the synchronized block.
+- A synchronized block may produce a `NullPointerException` if a parameter expression evaluates to null, while synchronized methods do not.
+- The lock is only held throughout the block scope of the synchronized block. 
+- A static method may lock an object inside the parenthesis of a synchronized block.
+
+### Implementing synchronization in Java
+To provide internal synchronization, Java's lock concept is employed. In Java, each object has its own lock. In this case, the lock idea will be brought into play anytime we utilize the synchronized keyword.
+
+First, a thread with the lock on the object must execute any synchronized methods on it. Post-lock, a thread may invoke any synchronized method on the object. Upon successful completion of the synchronized method, the thread is responsible for releasing the lock. No other threads may perform synchronized methods on the same object while a thread is executing one. However, any non-synchronized procedure may be executed concurrently by the remaining threads. Note that this idea of a lock may be applied on the object level rather than the method level.
+
+Let's look at an example program:
+
+```java
+class Implementation {
+
+	public synchronized void wish(String tag) {

```

Comment:
Inappropriate method name. It seems to be unrelated to the functionality of the method